cub/cub/device/dispatch/dispatch_scan.cuh

```diff
@@ -252,13 +253,12 @@ struct DispatchScan
   CUB_RUNTIME_FUNCTION _CCCL_HOST _CCCL_FORCEINLINE cudaError_t
   Invoke(InitKernelT init_kernel, ScanKernelT scan_kernel, ActivePolicyT policy = {})
   {
-    using ScanTileStateT = typename KernelSource::ScanTileStateT;
-
+    // TODO(ashwin): Does this now need to be a runtime check?
```

Comment:
Question for reviewers: I don't think we can do this at compile time anymore as `LoadModifier()` cannot always be `constexpr`. Should we change this to a runtime check? If so, should we throw a `runtime_error`?

Reply:
Sync'd offline and we decided to move this check to the policy. In C++, the check is a constexpr function call. In C, it happens at runtime.